server/controllers/jobs_controller.go

```diff
@@ -27,105 +36,24 @@ type JobsController struct {
 	Db                       *db.BoltDB
 	WsMux                    *websocket.Multiplexor
 	StatsScope               tally.Scope
-}
-
-type ProjectInfoKeyGenerator struct{}
-
-func (g ProjectInfoKeyGenerator) Generate(r *http.Request) (string, error) {
-	projectInfo, err := newProjectInfo(r)
-
-	if err != nil {
-		return "", errors.Wrap(err, "creating project info")
-	}
-
-	return projectInfo.String(), nil
-}
-
-type pullInfo struct {
-	org  string
-	repo string
-	pull int
-}
-
-func (p *pullInfo) String() string {
-	return fmt.Sprintf("%s/%s/%d", p.org, p.repo, p.pull)
-}
-
-type projectInfo struct {
-	projectName string
-	workspace   string
-	pullInfo
-}
-
-func (p *projectInfo) String() string {
-	return fmt.Sprintf("%s/%s/%d/%s/%s", p.org, p.repo, p.pull, p.projectName, p.workspace)
-}
-
-func newPullInfo(r *http.Request) (*pullInfo, error) {
-	org, ok := mux.Vars(r)["org"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no org in route")
-	}
-	repo, ok := mux.Vars(r)["repo"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no repo in route")
-	}
-	pull, ok := mux.Vars(r)["pull"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no pull in route")
-	}
-	pullNum, err := strconv.Atoi(pull)
-	if err != nil {
-		return nil, err
-	}
-
-	return &pullInfo{
-		org:  org,
-		repo: repo,
-		pull: pullNum,
-	}, nil
-}
-
-// Gets the PR information from the HTTP request params
-func newProjectInfo(r *http.Request) (*projectInfo, error) {
-	pullInfo, err := newPullInfo(r)
-	if err != nil {
-		return nil, err
-	}
-
-	project, ok := mux.Vars(r)["project"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no project in route")
-	}
-
-	workspace, ok := mux.Vars(r)["workspace"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no workspace in route")
-	}
-
-	return &projectInfo{
-		pullInfo:    *pullInfo,
-		projectName: project,
-		workspace:   workspace,
-	}, nil
+	KeyGenerator             websocket.PartitionKeyGenerator
```

Comment:
we don't need to use an interface defined in another package here since we're defining the implementation here, we should just reference `JobIDKeyGenerator` directly.

Reply:
Done!